content/client-lib-development-guide/features.textile

```diff
@@ -682,18 +693,17 @@ h4. ChannelOptions
 * @(TB1)@ options provided when instancing a channel, see "Java ChannelOptions":https://github.com/ably/ably-java/blob/master/src/io/ably/types/ChannelOptions.java as a reference
 * @(TB2)@ The attributes of @ChannelOptions@ consist of:
 ** @(TB2a)@ @encrypted@ boolean - when true, enables automatic encryption & decryption of all messages
-** @(TB2b)@ @cipherParams@ @CipherParams@ - when encrypted is true, the cipher params are required
+** @(TB2b)@ @cipherParams@ - This is required if @encrypted@ is true. It consists of either:
+*** @(TB2b1)@ A @CipherParams@ instance, or
+*** @(TB2b2)@ an options hash (or language equivalent) consisting of the key @key@ with value a key. In this case, the client library should call `getDefaultParams`, passing it the key, to obtain a @CipherParams@ instance
```

Comment:
I really can't say I like `Param[]`, for the reasons above. `getDefaultParams` is supposed to be a convenience method for the most usual case; if you are doing something more special than the default, you still can make your own `CipherParams`. I don't think we should complicate `getDefaultParams` for that case. (Plus, as @SimonWoolf notes above, it wouldn't really be "default" anymore.)

I'd rather restrict the hashmap convenience to dynamic languages and, as @SimonWoolf says, take just the key in those, or make your own `CipherParams`.